app/react/Layout/TableViewer.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import { createSelector } from 'reselect';
+
+import { connect } from 'react-redux';
+import { EntityDisplayState, IStore, TableViewColumn } from 'app/istore';
+import { Translate } from 'app/I18N';
+import { TableRows } from 'app/Layout/TableRows';
+import { CollectionViewerProps } from './CollectionViewerProps';
+
+export interface TableViewerProps extends CollectionViewerProps {
+  columns: TableViewColumn[];
+}
+
+class TableViewerComponent extends Component<TableViewerProps> {
+  handleScroll = (e: { target: any }) => {
+    const element = e.target;
+    if (element.scrollHeight - element.scrollTop === element.clientHeight) {
+      this.props.onEndScroll();
+    }
+  };
+
+  render() {
+    const columns = this.props.columns.filter((c: TableViewColumn) => !c.hidden);
+    return (
+      <div className="tableview-wrapper" onScroll={this.handleScroll}>
+        <table>
+          <thead>
+            <tr>
+              {columns.map((column: TableViewColumn, index: number) => (
+                <th className={!index ? 'sticky-col' : ''} key={column.name}>
+                  <div className="table-view-cell">
+                    <Translate>{column.label}</Translate>
+                  </div>
+                </th>
+              ))}
+            </tr>
+          </thead>
+          <tbody>
+            <TableRows
+              {...{
+                columns,
+                clickOnDocument: this.props.clickOnDocument,
+                storeKey: this.props.storeKey,
+                onEndScroll: this.props.onEndScroll,
+              }}
+            />
```

Comment:
Same as above, use the first line to assign variables to props, simplifies this greatly.